### PR TITLE
Implemented insert for Index & MultiIndex

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes `insert()` for `Index` and `MultiIndex`

```python
>>> kidx = ks.Index([1, 2, 3, 4, 5])
>>> kidx.insert(3, 100)
Int64Index([1, 2, 3, 100, 4, 5], dtype='int64')

>>> kidx = ks.Index([1, 2, 3, 4, 5])
>>> kidx.insert(-3, 100)
Int64Index([1, 2, 100, 3, 4, 5], dtype='int64')

>>> kmidx = ks.MultiIndex.from_tuples([("a", "x"), ("b", "y"), ("c", "z")])
>>> kmidx.insert(3, ("h", "j"))
MultiIndex([('a', 'x'),
            ('b', 'y'),
            ('c', 'z'),
            ('h', 'j')],
           )

>>> kmidx.insert(-2, ("h", "j"))
MultiIndex([('a', 'x'),
            ('h', 'j'),
            ('b', 'y'),
            ('c', 'z')],
                   )
```

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -2086,6 +2086,50 @@ def item(self):
         """
         return self.to_series().item()
 
+    def insert(self, loc: int, item):
+        """
+        Make new Index inserting new item at location.
+
+        Follows Python list.append semantics for negative values.
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:2093]: Let's don't say this. The order isn't guaranteed.

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -2086,6 +2086,50 @@ def item(self):
...
+        if loc < 0:
+            loc = loc + length
+            loc = 0 if loc < 0 else loc
+        else:
+            loc = length if loc > length else loc
+
+        index_name = self._internal.index_spark_column_names[0]
+        sdf = self._internal.spark_frame
+        sdf_before = self.to_frame(name=index_name)[:loc].to_spark()
+        sdf_middle = Index([item]).to_frame(name=index_name).to_spark()
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:2126]: Can we make series and/or index working as well?

File: databricks/koalas/indexes.py [hunk 3]
```diff
@@ -2086,6 +2086,50 @@ def item(self):
...
+        >>> kidx.insert(3, 100)
+        Int64Index([1, 2, 3, 100, 4, 5], dtype='int64')
+
+        For negative values
+
+        >>> kidx = ks.Index([1, 2, 3, 4, 5])
+        >>> kidx.insert(-3, 100)
+        Int64Index([1, 2, 100, 3, 4, 5], dtype='int64')
+        """
+        length = len(self)
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:2116]: If the given `loc` is positive, I think we don't have to calculate the total length here:

```python
>>> ks.Series([1,2,3])[:5]
0    1
1    2
2    3
dtype: int64
>>> ks.Series([1,2,3])[5:]
Series([], dtype: int64)
```

**Review by HyukjinKwon — APPROVED** [review]
Looks good otherwise.

File: databricks/koalas/indexes.py (hunk 3 above)
Comment (itholic) [reply databricks/koalas/indexes.py]: Oh, thanks!

**Pull request merged**